assets/js/modules/analytics/components/module/ModuleOverviewWidget/SiteStats.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * SiteStats component the ModuleOverviewWidget widget.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import { extractAnalyticsDashboardData } from '../../../util';
+import { getCurrentDateRangeDayCount } from '../../../../../util/date-range';
+import GoogleChart from '../../../../../components/GoogleChart';
+import PreviewBlock from '../../../../../components/PreviewBlock';
+import { Cell, Row, Grid } from '../../../../../material-components';
+const { useSelect } = Data;
+
+export default function SiteStats( { loaded, selectedStat, report } ) {
+	const dateRange = useSelect( ( select ) => select( CORE_USER ).getDateRange() );
+
+	if ( ! loaded ) {
+		return (
+			<PreviewBlock width="100%" height="270px" />
+		);
+	}
+
+	const currentDayCount = getCurrentDateRangeDayCount( dateRange );
+	const dataMap = extractAnalyticsDashboardData( report, {
+		selectedStats: selectedStat,
+		selectedDataIndex: 0,
+		currentMonthMetricIndex: 0,
+		previousMonthMetricIndex: 1,
+		days: currentDayCount,
+	} );
```

Comment:
Instead of using and modifying this legacy function, let's come up with a more flexible (or at least more understandable) way to do that. Like in the other widget components implemented for the Site Kit dashboard, we shouldn't use old functions that serve a super-specific purpose. Let's introduce some less specific utility functions instead, and the parts that need to be specific to the `report` used here should be kept within this component.

Essentially, `extractAnalyticsDashboardData` can be broken down into multiple, parameter-controlled functions where possible. The new component should then use a mix of those flexible functions while hard-coding e.g. `dataLabels` in the component since it's specific to that.

Reply:
@felixarntz this `extractAnalyticsDashboardData` is only intended for the legacy and new site stats components. We don't use it anywhere else. So I don't think we need to over-engineering here and introduce a more complicated approach.

I have updated it to minimize the number of changes in that function to make it less complicated. Let me know if it works for you.

Reply:
Left a follow-up comment on my new review.